### PR TITLE
Add Google Cloud Bigtable module.

### PR DESCRIPTION
**Pull request by mbrukman**
This PR resolves issue #587.

Thank you for contributing to JanusGraph.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [x] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

/cc: @lindsayismith, @igorbernstein2, @sduskis

### CODE REVIEW

File: janusgraph-bigtable/pom.xml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    <properties>
+        <top.level.basedir>${basedir}/..</top.level.basedir>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>com.google.cloud.bigtable</groupId>
+            <artifactId>bigtable-hbase-1.x-shaded</artifactId>
+            <version>1.0.0-pre3</version>
```
Comment (igorbernstein) [inline janusgraph-bigtable/pom.xml]: Might be good to extract the versions into properties in the root pom.xml

Comment (mbrukman) [reply janusgraph-bigtable/pom.xml]: Done; PR updated and re-pushed.

**Comment by igorbernstein2** [comment]
LGTM

File: janusgraph-bigtable/pom.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                  <exclusion>
+                       <groupId>org.apache.hbase</groupId>
+                       <artifactId>hbase-shaded-client</artifactId>
+                  </exclusion>
+            </exclusions>
+        </dependency>
+
+        <dependency>
+           <groupId>io.netty</groupId>
+           <artifactId>netty-tcnative-boringssl-static</artifactId>
```
Comment (jerryjch) [inline janusgraph-bigtable/pom.xml]: Is this dependency needed here?  From the links here, it does not seem to be.
https://cloud.google.com/bigtable/docs/using-maven
https://cloud.google.com/bigtable/docs/upgrading-clients

Comment (igorbernstein2) [reply janusgraph-bigtable/pom.xml]: You are right, it's no longer necessary.

Comment (mbrukman) [reply janusgraph-bigtable/pom.xml]: Removed `netty-tcnative-boringssl-static`.

**Comment by pluradj** [comment]
Is the [version support matrix](https://github.com/JanusGraph/janusgraph/blob/master/docs/versions.txt#L25) for still correct after this change?

```0.9.z, 1.0.0-preZ```

**Comment by pluradj** [comment]
Thanks @mbrukman , I was wondering more about the `0.9.z` version because if support for that dropped out, we'd need migration notes.

**Comment by sduskis** [comment]
It depends on what you mean by support.  The Cloud Bigtable team will help with any issues that come up with 0.9.7.z.  FWIW, 1.0.0-pre1 - pre3 are drop in replacements for 0.9.7.z from the perspective of API and dependencies.

**Comment by pluradj** [comment]
Meaning JanusGraph will still work with Cloud Bigtable `0.9.z`. Sounds like it will, so I don't think any update is required to the matrix. Thanks.

**Comment by mbrukman** [comment]
Thanks for the clarification, @sduskis!

@pluradj, any other questions/comments on the PR? We need one more committer approval before merging.

**Pull request merged**